### PR TITLE
boards/nucleo-f207zg:  support for ADC

### PR DESCRIPTION
**Pull request by krzysztof-cabaj**
### Contribution description

This PR adds to the Nucleo-f207zg ADC configuration.

### Testing procedure

Flash the board using `tests/periph_adc` program. Check if measured values changes when A0-A5 pins are
connected to the 3,3V, GND or to the potentiometer. 

### Issues/PRs references

None

### CODE REVIEW

File: cpu/stm32/include/periph/f2/periph_cpu.h [hunk 1]
```diff
@@ -28,7 +28,11 @@ extern "C" {
 /**
  * @brief   Available number of ADC devices
  */
+#if defined(CPU_MODEL_STM32F207ZG)
```
Comment (benpicco) [inline cpu/stm32/include/periph/f2/periph_cpu.h:31]: Does this work?
```suggestion
#if defined(ADC3)
```

Comment (krzysztof-cabaj) [reply cpu/stm32/include/periph/f2/periph_cpu.h]: I checked both solutions on hardware and both are working. I commit your change. 
BTW. Where in stm32 code is `ADC3` defined?

File: cpu/stm32/include/periph/f2/periph_cpu.h [hunk 2]
```diff
@@ -28,7 +28,12 @@ extern "C" {
 /**
  * @brief   Available number of ADC devices
  */
+//#if defined(CPU_MODEL_STM32F207ZG)
```
Comment (benpicco) [inline cpu/stm32/include/periph/f2/periph_cpu.h:31]:
```suggestion
```

File: cpu/stm32/include/periph/f2/periph_cpu.h (hunk 1 above)
Comment (benpicco) [reply cpu/stm32/include/periph/f2/periph_cpu.h]: > Where in stm32 code is `ADC3` defined?

Should be in `cpu/stm32/include/vendor`

Comment (krzysztof-cabaj) [reply cpu/stm32/include/periph/f2/periph_cpu.h]: Thanks! And sorry for some old, test code ;)

**Comment by benpicco** [comment]
Please squash, you only need two commits: one for the CPU and one for the board.

**Comment by krzysztof-cabaj** [comment]
Squashed to two commits.

**Review by benpicco — APPROVED** [review]

**Pull request merged**